bundles/action/org.openhab.action.pushover/src/main/java/org/openhab/action/pushover/internal/Pushover.java

```diff
@@ -399,6 +412,254 @@ private static String parseResponse(String response)
+                emergencyPriority, sound);
+    }
+
+    // Primary method for sending a message to the Pushover API
+    @ActionDoc(text = "Send a notification to your Android device. apiKey, user and message are required. All else can effectively be null.", returns = "a <code>receipt</code> (30 character string containing the character set [A-Za-z0-9]), if successful and <code>null</code> otherwise.")
+    public static String sendPushover(@ParamDoc(name = "apiKey", text = "Your application's API token.") String apiKey,
+            @ParamDoc(name = "user", text = "The user/group key (not e-mail address) of your user (or you), viewable when logged into Pushover dashboard.") String user,
+            @ParamDoc(name = "message", text = "Your message.") String message,
+            @ParamDoc(name = "device", text = " Your user's device name to send the message directly to that device, rather than all of the user's devices.") String device,
+            @ParamDoc(name = "title", text = "Your message's title, otherwise your app's name is used.") String title,
+            @ParamDoc(name = "url", text = "A supplementary URL to show with your message.") String url,
+            @ParamDoc(name = "urlTitle", text = "A title for your supplementary URL, otherwise just the URL is shown.") String urlTitle,
+            @ParamDoc(name = "priority", text = "Send as -1 to always send as a quiet notification, 1 to display as high-priority and bypass the user's quiet hours, or 2 to also require confirmation from the user.") int priority,
+            @ParamDoc(name = "sound", text = "The name of one of the sounds supported by device clients to override the user's default sound choice.") String sound) {
+
+        StringBuilder data = new StringBuilder();
+
+        try {
+            if (!StringUtils.isEmpty(apiKey)) {
+                addEncodedParameter(data, MESSAGE_KEY_API_KEY, apiKey);
+            } else if (!StringUtils.isEmpty(defaultApiKey)) {
+                addEncodedParameter(data, MESSAGE_KEY_API_KEY, defaultApiKey);
+            } else {
+                logger.error("Application API token not specified.");
+                return null;
+            }
+
+            if (!StringUtils.isEmpty(user)) {
+                addEncodedParameter(data, MESSAGE_KEY_USER, user);
+            } else if (!StringUtils.isEmpty(defaultUser)) {
+                addEncodedParameter(data, MESSAGE_KEY_USER, defaultUser);
+            } else {
+                logger.error("The user/group key was not specified.");
+                return null;
+            }
+
+            if (!StringUtils.isEmpty(message)) {
+                if ((message.length() + title.length()) <= API_MAX_MESSAGE_LENGTH) {
+                    addEncodedParameter(data, MESSAGE_KEY_MESSAGE, message);
+                } else {
+                    logger.error("Together, the event message and title total more than " + API_MAX_MESSAGE_LENGTH
```

Comment:
This should be a WARN instead of ERROR.
Also, please convert to parameterized logging.